lib/kafka/consumer.rb

```diff
@@ -212,6 +212,8 @@ def each_message(min_bytes: 1, max_bytes: 10485760, max_wait_time: 1, automatica
             })
           end
           batch.messages.each do |message|
+            next if paused?(message.topic, message.partition)
```

Comment:
This will effectively _skip_ all messages while a partition is paused, which is not the desired behavior.